uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Helper/ContactActionsMenu/LeaveActionGroup.php

```diff
@@ -0,0 +1,230 @@
+    $this->leaveManagerService = $leaveManagerService;
+    $this->contactID = $contactID;
+  }
+
+  /**
+   * Gets Leave Menu Group with menu items already
+   * added.
+   *
+   * @return ActionsGroup
+   */
+  public function get() {
+    $actionsGroup = new ActionsGroup('Leave:');
+    $actionsGroup->addItem($this->getRecordLeaveButton());
+    $actionsGroup->addItem($this->getRecordSicknessButton());
+    $actionsGroup->addItem($this->getRecordOvertimeButton());
+    $actionsGroup->addItem($this->getViewEntitlementsButton());
+    $actionsGroup->addItem(new GroupSeparatorItem());
+
+    $leaveApprovers = $this->getLeaveApprovers();
+
+    if ($leaveApprovers) {
+      $leaveApproversListItem = new LeaveApproversListItem($this->leaveManagerService, $this->contactID);
+      $actionsGroup->addItem($leaveApproversListItem);
+      $actionsGroup->addItem($this->getManageLeaveApproverButton());
+    }
+    else {
+      $noLeaveApprovertItem = new NoSelectedLeaveApproverItem();
+      $actionsGroup->addItem($noLeaveApprovertItem);
+      $actionsGroup->addItem($this->getAddLeaveApproverButton());
+    }
+
+    return $actionsGroup;
+  }
+
+  /**
+   * Gets the View Entitlement button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getViewEntitlementsButton() {
+    $params = [
+      'label' => 'View Entitlements',
+      'class' => 'btn-primary-outline',
+      'icon' => 'fa-search',
+      'url' => $this->getLeaveTabUrl()
+    ];
+
+    return $this->getMenuButton($params);
+  }
+
+  /**
+   * Gets the Add Leave Approver button.
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getAddLeaveApproverButton() {
+    $relTypeID = $this->getLeaveApproverRelationshipTypeSelectId();
+    $attribute = ['onclick' => "CRM.loadForm('/civicrm/contact/view/rel?cid={$this->contactID}&action=add&relTypeId={$relTypeID}')"];
+    $params = [
+      'label' => 'Add Leave Approver',
+      'class' => 'btn-secondary-outline',
+      'icon' => '',
+      'url' => '#'
+    ];
+
+    return $this->getMenuButton($params, $attribute);
+  }
+
+  /**
+   * Gets the Manage Leave Approver button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getManageLeaveApproverButton() {
+    $url = CRM_Utils_System::url(
+      'civicrm/contact/view',
+      "reset=1&cid={$this->contactID}&selectedChild=rel"
+    );
+    $params = [
+      'label' => 'Manage Leave Approver',
+      'class' => 'btn btn-secondary',
+      'icon' => '',
+      'url' => $url
+    ];
+
+    return $this->getMenuButton($params);
+  }
+
+  /**
+   * Gets the Record Sickness button.
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getRecordSicknessButton() {
+    return $this->getLeaveButton('Record Sickness', 'fa-stethoscope', 'sickness');
+  }
+
+  /**
+   * Gets the Record Leave button.
+   *
+   * @return \CRM_HRContactActionsMenu_Component_GroupButtonItem
```

Comment:
Any reason this `@return` is different from the other button methods?

Reply:
Nope.